### PR TITLE
Print additional information if MemberInfo is a method

### PR DESCRIPTION
**Pull request by TermanEmil**
I created the `IMemberInfoDescriptor` which is implemented by `MethodInfoDescriptor` and `PropertiesDescriptor`.

To be able to test the output, the `IMemberInfoDescriptor.Describe()` returns `IEnumerable<string>`.

### CODE REVIEW

File: src/DotNetEssentials/ObjectBrowser/MetadataMenu/MethodInfoDescriptor.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        private IEnumerable<string> GetMethodInfoParametersStr(MethodInfo methodInfo)
+        {
+            var parameters = methodInfo.GetParameters();
+
+            if (!parameters.Any())
+                yield break;
+
+            yield return "Parameters:";
+
+            var isFirstParameter = false;
```
Comment (maksimenko-stanislav) [inline src/DotNetEssentials/ObjectBrowser/MetadataMenu/MethodInfoDescriptor.cs]: Did you mean `var isFirstParameter = true`?

File: src/DotNetEssentials/ObjectBrowser/MetadataMenu/MethodInfoDescriptor.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            {
+                yield return MemberInfoShortDescriptor.GetMemberInfoString(member);
+            }
+        }
+
+        private string GetParameterDescription(ParameterInfo parameter)
+        {
+            var builder = new StringBuilder().Append(parameter.ParameterType).Append(' ');
+
+            if (parameter.IsOut) builder.Append("out ");
```
Comment (maksimenko-stanislav) [inline src/DotNetEssentials/ObjectBrowser/MetadataMenu/MethodInfoDescriptor.cs]: Nice check!

File: src/DotNetEssentials/ObjectBrowser/ObjectBrowserTests/MetadataMenu/MethodInfoDescriptorTests.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
+﻿using System;
```
Comment (maksimenko-stanislav) [inline src/DotNetEssentials/ObjectBrowser/ObjectBrowserTests/MetadataMenu/MethodInfoDescriptorTests.cs]: Nice to have unit tests!

File: src/DotNetEssentials/ObjectBrowser/MetadataMenu/ConsoleMenuExtensions.cs [hunk 4]
```diff
@@ -67,53 +65,45 @@ private static ActionMenuCommand GetMemberMenuItem(this MemberInfo memberInfo, I
...
-				}
-			}
-
-			return $"{memberInfo.MemberType} {memberInfo.Name}";
-		}
-	}
+            Console.WriteLine(MemberInfoShortDescriptor.GetMemberInfoString(memberInfo));
+
+            // This should be initialized somewhere else
+            var descriptors = new IMemberInfoDescriptor[] { new MethodInfoDescriptor(), new PropertiesDescriptor() };
```
Comment (maksimenko-stanislav) [inline src/DotNetEssentials/ObjectBrowser/MetadataMenu/ConsoleMenuExtensions.cs]: What about fields? It is required to implement IMemberInfoDescriptor for it?

**Comment by TermanEmil** [comment]
I've just realized, that logic with isFirstParameter is wrong altogether

**Comment by TermanEmil** [comment]
Yes, why not?

**Comment by TermanEmil** [comment]
> What about fields? It is required to implement IMemberInfoDescriptor for
it?

Yes, why not?

File: src/DotNetEssentials/ObjectBrowser/MetadataMenu/MethodInfoDescriptor.cs (hunk 1 above)
Comment (TermanEmil) [reply src/DotNetEssentials/ObjectBrowser/MetadataMenu/MethodInfoDescriptor.cs]: I think the logic is wrong altogether. I should just `yield return ""` and remove `isFirstParameter`

File: src/DotNetEssentials/ObjectBrowser/MetadataMenu/ConsoleMenuExtensions.cs (hunk 4 above)
Comment (TermanEmil) [reply src/DotNetEssentials/ObjectBrowser/MetadataMenu/ConsoleMenuExtensions.cs]: Yes. Is there a problem with this approach?

**Pull request closed without merging**